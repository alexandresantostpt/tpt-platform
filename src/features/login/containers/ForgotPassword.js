import { connect } from 'react-redux'

import React from 'react'

import { requestRememberPasswordAction } from '@features/login/actions'

import GlobalStyle from '@components/globalStyle/GlobalStyle'
import Toasts from '@components/toast/Toasts'

import ForgotPassword from '../pages/ForgotPassword'

const ForgotPasswordContainer = props => (
    <>
        <GlobalStyle />
        <Toasts />
        <ForgotPassword {...props} />
    </>
)

const mapDispatchToProps = {
    requestRememberPasswordAction
}

export default connect(
    null,
    mapDispatchToProps
)(ForgotPasswordContainer)
