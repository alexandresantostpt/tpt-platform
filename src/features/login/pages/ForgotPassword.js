import PropTypes from 'prop-types'
import React, { useState } from 'react'

import { i18n } from '@i18n'

import Form from '../components/form/Form'
import Button from '../components/button/Button'
import ErrorMessage from '../components/errorMessage/ErrorMessage'
import Input from '../components/input/Input'
import LinkTo from '../components/link/LinkTo'
import LoginComponent from '../components/login/Login'
import Text from '../components/text/Text'

import { routesLogin } from '../routes'

const ForgotPassword = props => {
    const [email, updateEmail] = useState('')
    const [notEqual, updateNotEqual] = useState(false)

    const sendButtonClicked = event => {
        event.preventDefault()
        const { requestRememberPasswordAction } = props

        if (email !== '') {
            updateNotEqual(false)
            requestRememberPasswordAction({ email })
        } else {
            updateNotEqual(true)
        }
    }

    return (
        <LoginComponent>
            <Text>
                <div>{i18n.t('titles.welcome')}</div>
                <div>{i18n.t('titles.typeYourEmail')}</div>
            </Text>
            <Form onSubmit={sendButtonClicked}>
                <ErrorMessage>{notEqual ? i18n.t('messages.errors.typeTheEmail') : ''}</ErrorMessage>
                <Input id={'email'} onChange={updateEmail} placeholder={i18n.t('placeholders.email')} type="text" />
                <Button disabled={false} text={i18n.t('buttons.send')} type="submit" />
                <LinkTo text={i18n.t('link.cancel')} to={routesLogin.path} />
            </Form>
        </LoginComponent>
    )
}

ForgotPassword.propTypes = {
    requestRememberPasswordAction: PropTypes.func.isRequired
}

export default ForgotPassword
