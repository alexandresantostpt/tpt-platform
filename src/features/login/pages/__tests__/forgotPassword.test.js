import React from 'react'
import renderer from 'react-test-renderer'

import ForgotPassword from '../ForgotPassword'

describe('Tests for ForgotPassword component', () => {
    it('Should render withou crash', () => {
        const tree = renderer.create(<ForgotPassword requestRememberPasswordAction={jest.fn()} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
