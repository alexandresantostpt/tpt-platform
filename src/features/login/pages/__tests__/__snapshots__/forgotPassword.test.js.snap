// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for ForgotPassword component Should render withou crash 1`] = `
<main
  className="Page__StyledPage-sc-9alblw-0 eBjaLI"
>
  <div
    className="Background__StyledBg-sc-904t04-0 bTqgFQ"
  />
  <div
    className="Box__StyledBox-v5mj6b-0 llPnpv"
  >
    <div
      className="Text__StyledText-fa81pq-0 bIShHK"
    >
      <div>
        Bem-vindo!
      </div>
      <div>
        Digite seu email abaixo
      </div>
    </div>
    <form
      className="Form__StyledForm-sc-2b0hvf-0 hgpEDk"
      onSubmit={[Function]}
    >
      <div
        className="ErrorMessage__StyledDiv-maht7p-0 faewLc"
      >
        
      </div>
      <div
        className="Input__StyledBox-sc-4bhgmp-0 gcStMN"
      >
        <input
          className="Input__StyledInput-sc-4bhgmp-3 hoXjZP"
          id="email"
          onChange={[Function]}
          required={true}
          type="text"
        />
        <label
          className="Input__StyledLabel-sc-4bhgmp-2 lcosGj"
          htmlFor="email"
        >
           
          Email
           
        </label>
      </div>
      <button
        className="Button__StyledButton-sc-6tss9n-0 dHMzfC"
        disabled={false}
        type="submit"
      >
        Enviar
      </button>
      <a
        className="LinkTo__StyledLink-sc-1u70co-0 khxOGt"
        to="/login"
      >
        Cancelar
      </a>
    </form>
  </div>
</main>
`;
